demux/demux_lavf.c

```diff
@@ -1005,6 +1006,9 @@ static int demux_open_lavf(demuxer_t *demuxer, enum demux_check check)
                    "analyzeduration to %f\n", analyze_duration);
     }
 
+    if (priv->format_hack.no_ext_picky)
+        av_dict_set(&dopts, "extension_picky", "0", 0);
```

Comment:
I didn't read too closely why we mix options in avdict and in avformat directly, but how about we use `av_opt_set_int(avfc, ...`? Or `av_dict_set_int` for that matter, but not sure why use dict.

Reply:
You can't set demuxer private options from AVFormatContext, only from the options arg of `avformat_open_input`. Which is also why I initially didn't put this in `format_hacks`. Because format_hacks don't have any demuxer private options.

I can change it to av_dict_set_int in any case.

Reply:
> You can't set demuxer private options from AVFormatContext

I think you can with `AV_OPT_SEARCH_CHILDREN`, but we do it before open, so that's correct to do in dict.

> I can change it to av_dict_set_int in any case.

Actually sorry, in this case, we might want to keep it as string "0", dict is string->string map anyway, so it would convert anyway. (unlike av_opt_set_int which would set int directly).